ansible/playbooks/roles/redpanda_broker/templates/configs/defaults.j2

```diff
@@ -19,13 +31,25 @@ node:
     kafka_api:
     - address: {{ hostvars[inventory_hostname].private_ip }}
       port: {{ redpanda_kafka_port }}
+{% if advertise_public_ips is defined %}
+{% if advertise_public_ips %}
+      name: internal
+{% endif %}
+{% endif %}
+{% if advertise_public_ips is defined %}
+{% if advertise_public_ips %}
+    - address: 0.0.0.0
+      port: {{ redpanda_external_kafka_port }}
+      name: external
+{% endif %}
+{% endif %}
     admin:
     - address: {{ hostvars[inventory_hostname].private_ip }}
       port: {{ redpanda_admin_api_port }}
     seed_servers:
 {% for host in groups["redpanda"] %}
       - host:
-        address: {{ hostvars[host]['inventory_hostname' if (hostvars[host].advertise_public_ips | d() | bool) else 'private_ip'] }}
```

Comment:
seed servers does not seem to allow for public ip here

Reply:
+1